wal/metrics.go

```diff
@@ -16,13 +16,20 @@ package wal
 
 import "github.com/prometheus/client_golang/prometheus"
 
+const (
+	maxBucket                    = 16384
+	linearBucketWidthForSlowSync = 2
+)
+
 var (
+	// We support a specific metric for counting warnings since that is a commonly observed metric for etcd failures.
+	// Ideally the biggest buckets will be in the lower ranges.  If the larger buckets are growing, etcd disks maybe be full/too slow.
 	syncDurations = prometheus.NewHistogram(prometheus.HistogramOpts{
 		Namespace: "etcd",
 		Subsystem: "disk",
 		Name:      "wal_fsync_duration_seconds",
 		Help:      "The latency distributions of fsync called by wal.",
-		Buckets:   prometheus.ExponentialBuckets(0.001, 2, 14),
+		Buckets:   prometheus.LinearBuckets(0.001, linearBucketWidthForSlowSync, maxBucket/linearBucketWidthForSlowSync),
```

Comment:
I tried this, these metrics probably aren't useful since they add 2 seconds to each bucket:
```
etcd_disk_wal_fsync_duration_seconds_bucket{le="0.001"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="2.001"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="4.0009999999999994"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="6.0009999999999994"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="8.001"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="10.001"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="12.001"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="14.001"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="16.000999999999998"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="18.000999999999998"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="20.000999999999998"} 3
etcd_disk_wal_fsync_duration_seconds_bucket{le="22.000999999999998"} 3
...
```

What sort of granularity are you expecting here and does it really need 8k buckets? 96% of all metrics end up being fsync metrics
```sh
$ curl localhost:2379/metrics | grep fsync | wc -l
8197
$ curl localhost:2379/metrics | grep fsync | wc -l
8519
```